sqe-api-server/Helpers/InputValidation.cs

```diff
@@ -71,32 +63,38 @@ public static bool ValidateTransformMatrix(string transformMatrix)
         }
 
         /// <summary>
-        /// Validate that the wkt polygon is indeed correct. If it is not, the method will
-        /// throw an error. When the fix parameter is set to true, it will try to repair it and
-        /// send the repaired version of the polygon back with the error.  If the polygon cannot be
-        /// repaired at all, then a descriptive error about the polygon is thrown.
+        ///     Validate that the wkt polygon is indeed correct. If it is not, the method will
+        ///     throw an error. When the fix parameter is set to true, it will try to repair it and
+        ///     send the repaired version of the polygon back with the error.  If the polygon cannot be
+        ///     repaired at all, then a descriptive error about the polygon is thrown.
         /// </summary>
         /// <param name="wktPolygon">A wkt polygon</param>
         /// <param name="entityName">The type of object being validated (used in formulating useful exception error messages)</param>
-        /// <param name="fix">Invalid polygons always throw an error, this flag determines whether to try to return a repaired
-        /// version of the polygon with that error</param>
+        /// <param name="fix">
+        ///     Invalid polygons always throw an error, this flag determines whether to try to return a repaired
+        ///     version of the polygon with that error
+        /// </param>
         /// <returns>A WKT string with the cleaned polygon</returns>
         public static async Task<string> ValidatePolygonAsync(string wktPolygon, string entityName, bool fix = false)
         {
             // Wrap the private validation method in a Task so we can asynchronously call this non-trivial method
+            // But see the docs: https://docs.microsoft.com/en-us/aspnet/core/performance/performance-best-practices?view=aspnetcore-3.1
```

Comment:
I'm not sure why they warn against it, as it's not explained. Perhaps they're worried about exhausting the threadpool? Actually, that may be it. You have 20 threads, one of them is handling the current request. You can call the lengthy method on that thread, or you do a Task.Run, free up your thread but make another one block. And you're request is still waiting. All you've done with Task.Run, is add a little overhead. I guess that's the reason.

I'm not sure what the right solution is here. Starting up a Polygon Validation Service that manages its own threads just seem like an overkill here. Perhaps just don't use Task.Run at all, and we'll see if validatePolygon is slowing us down - then we'll worry about it?

https://social.msdn.microsoft.com/Forums/vstudio/en-US/1124354e-7365-430e-9bc4-018138e0d601/creating-multiple-threadpools?forum=netfxbcl - this contains an explanation why lengthy calculations should not be done in thread pool threads.

Reply:
I agree, I think we just remove Task.Run(). From reading the various discussions it is unclear what the best approach is. I understand that one doesn't want a long-running process holding up the thread, but it has to run somewhere and perhaps a dedicated service is the "proper" way to handle it. I guess I will just remove the Task.Run() and leave the comments there.  Thanks for taking a look at it.